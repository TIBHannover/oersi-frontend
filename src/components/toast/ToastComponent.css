.toast__container {
  display: table-cell;
  vertical-align: middle;
  background-color: transparent;
  color: white;
}

.toast__cell {
  display: inline-block;
}

.add-margin {
  margin-top: 20px;
}

.toast__svg {
  fill: #fff;
}

.toast {
  text-align: left;
  border-radius: 4px;
  max-width: 500px;
  top: 0px;
  position: relative;
}

.toast__icon {
  position: absolute;
  top: 50%;
  left: 22px;
  transform: translateY(-50%);
  width: 14px;
  height: 14px;
  padding: 7px;
  border-radius: 50%;
  display: inline-block;
  background-color: yellow;
}

.toast__type {
  font-weight: 700;
  margin-top: 0;
  margin-bottom: 8px;
}

.toast__message {
  font-size: 14px;
  margin-top: 0;
  margin-bottom: 0;
}

.toast__content {
  padding-left: 70px;
  padding-right: 60px;
}

.toast__close {
  position: absolute;
  right: 22px;
  top: 50%;
  width: 14px;
  cursor: pointer;
  height: 14px;
  fill: #d2d2d2;
  transform: translateY(-50%);
}

.toast--green .toast__icon {
  background-color: #2bde3f;
}

.toast--green:before {
  background-color: #2bde3f;
}
