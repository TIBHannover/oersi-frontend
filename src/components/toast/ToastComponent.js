import React from "react"
import {toast, Zoom} from "react-toastify"
import "./ToastComponent.css"

const ToastComponent = (props) => {
  return <>{notify({...props})}</>

  function notify({title, message, type}) {
    return toast(
      <div className="toast__container">
        <div className="toast__cell">
          <div className="">
            <div className="toast__icon"></div>
            <div className="toast__content">
              <p className="toast__type">{title}</p>
              <p className="toast__message">{message}</p>
            </div>
            <div className="toast__close"></div>
          </div>
        </div>
      </div>,
      {
        transition: Zoom,
        type: type,
        autoClose: 5000,
      }
    )
  }
}
export default ToastComponent
