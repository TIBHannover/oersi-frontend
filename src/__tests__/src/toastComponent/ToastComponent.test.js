import React from "react"
import ReactDOM from "react-dom"
import ToastComponent from "../../../components/toast/ToastComponent"

let container = null
beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement("div")
  document.body.appendChild(container)
})

describe("ToastComponent ==> Test UI  ", () => {
  it("ToastComponent : should render without crashing", async () => {
    const div = document.createElement("div")
    ReactDOM.render(
      <ToastComponent
        message={"Test message "}
        title={"Test title"}
        type={"success"}
      />,
      div
    )

    ReactDOM.unmountComponentAtNode(div)
  })
})
