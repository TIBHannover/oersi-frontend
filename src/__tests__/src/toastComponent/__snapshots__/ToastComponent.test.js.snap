// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToastComponent ==> Test UI   ToastComponent : should render correctly 1`] = `ShallowWrapper {}`;
